src/stores/A_store.js

```diff
@@ -155,30 +155,30 @@ class A_store extends ReduceStore {
 
             case ActionTypes.DSTARLITE: {
                 const {player, end, pathGrid} = state;
-                const {grid} = pathGrid;
-                const DSL = new Dstarlite(grid[player.y][player.x], grid[end.y][end.x], Cell.heuristic);
-                DSL.beginPathfinding();
+                const dsl = new Dstarlite(pathGrid.getCell(player.x, player.y), pathGrid.getCell(end.x, end.y), Cell.heuristic);
+                dsl.beginPathfinding();
 
-                return {...state, DSL, stage: 'DSTAR_STEP'};
+                return {...state, dsl, stage: 'DSTAR_STEP'};
             }
 
             case ActionTypes.DSTAR_STEP: {
-                const {DSL, seenCells, revealedGrid, end} = state;
+                const {dsl, seenCells, revealedGrid, end} = state;
+                const nextLocation = dsl.nextStep();
                 let {pathGrid} = state;
-                const nextLocation = DSL.nextStep();
                 let stage;
                 let visibleCells;
+                let changedCells;
 
-                if (nextLocation.x !== end.x || nextLocation.y !== end.y) {
+                if (nextLocation.x === end.x && nextLocation.y === end.y || nextLocation === null) {
```

Comment:
if you ever have a mix of `&&` and `||` [plis parenthesize](https://eslint.org/docs/rules/no-mixed-operators)

Reply:
🐇